client/core/core.go

```diff
@@ -2632,6 +2633,84 @@ func (c *Core) Trade(pw []byte, form *TradeForm) (*Order, error) {
 	return corder, nil
 }
 
+// AccountExport is used to retrieve account by host for export.
+func (c *Core) AccountExport(pw []byte, host string) (*Account, error) {
+	_, err := c.encryptionKey(pw)
+	if err != nil {
+		return nil, codedError(passwordErr, err)
+	}
+	_, err = addrHost(host)
+	if err != nil {
+		return nil, newError(addressParseErr, "error parsing address: %v", err)
+	}
+
+	// Get the dexConnection and the dex.Asset for each asset.
+	c.connMtx.RLock()
+	dc, found := c.conns[host]
+	c.connMtx.RUnlock()
+	if !found {
+		return nil, newError(unknownDEXErr, "DEX: %s", host)
+	}
+	account := &Account{
+		Host:      host,
+		AccountID: dc.acct.id.String(),
+		PrivKey:   hex.EncodeToString(dc.acct.privKey.Serialize()),
+		PubKey:    hex.EncodeToString(dc.acct.dexPubKey.SerializeUncompressed()),
+		EncKey:    hex.EncodeToString(dc.acct.encKey),
+		Cert:      hex.EncodeToString(dc.acct.cert),
+		FeeCoin:   hex.EncodeToString(dc.acct.feeCoin),
+		IsPaid:    dc.acct.isPaid,
+	}
+	return account, nil
+}
+
+// AccountImport is used import an existing account into the db.
+func (c *Core) AccountImport(pw []byte, account Account) error {
+	_, err := c.encryptionKey(pw)
+	if err != nil {
+		return codedError(passwordErr, err)
+	}
+	_, err = addrHost(account.Host)
+	if err != nil {
+		return newError(addressParseErr, "error parsing address: %v", err)
+	}
+	var accountInfo db.AccountInfo
+	accountInfo.Host = account.Host
+	pubKey, err := hex.DecodeString(account.PubKey)
+	if err != nil {
+		return codedError(decodeErr, err)
+	}
+	accountInfo.DEXPubKey, err = secp256k1.ParsePubKey(pubKey)
+	if err != nil {
+		return codedError(parseKeyErr, err)
+	}
+	encKey, err := hex.DecodeString(account.EncKey)
+	if err != nil {
+		return codedError(decodeErr, err)
+	}
+	accountInfo.EncKey = encKey
+	cert, err := hex.DecodeString(account.Cert)
+	if err != nil {
+		return codedError(decodeErr, err)
+	}
+	accountInfo.Cert = cert
+	feeCoin, err := hex.DecodeString(account.FeeCoin)
+	if err != nil {
+		return codedError(decodeErr, err)
+	}
+	accountInfo.FeeCoin = feeCoin
+	if !c.verifyAccount(&accountInfo) {
+		return newError(accountVerificationErr, "Account not verified for host: "+account.Host, err)
+	}
+	accountInfo.Paid = c.conns[account.Host].acct.isPaid
```

Comment:
The c.verifyAccount() method populates c.conns.

I would think that you want IsPaid to come from the value known from the dex. Otherwise the user could just change the value in the file. I'm not sure of the implications of them faking that value into the db.

Is your point that the user would have no motivation to fake data since the dex isPaid value is always verified? Or in other words we don't have to trust any client data, it exists purely for convenience?

Reply:
Ah, no I just neglected verifyAccount when I made the comment.